app/models/mixins/authentication_mixin.rb

```diff
@@ -273,6 +273,27 @@ def authentication_check(*args)
     return status == :valid, details
   end
 
+  def authentication_check_queue(*args)
```

Comment:
If there's something you can't do, please provide a test or script demonstrating what's missing and I'll try to fix the code to do what you want.